astropy/table/table.py

```diff
@@ -3030,6 +3030,29 @@ def reverse(self):
         for index in self.indices:
             index.reverse()
 
+    def round(self, decimals=0):
+        '''
+        Round columns to a specific number of decimals.
+
+        Parameters
+        ----------
+        decimals: int, dict
+            Number of decimals to round the columns to. If a dict is given,
+            the columns will be rounded to the number specified as the value.
+            If a certain column is not in the dict given, it will remain the
+            same.
+        '''
+        if isinstance(decimals, dict):
```

Comment:
One thing I realize is that any numeric type like int, complex, byte columns can also be valid columns for rounding, for instance if you specify `decimals=-1` then that rounds to the nearest 10.  See e.g. https://numpy.org/devdocs/reference/arrays.dtypes.html#arrays-dtypes

Another thing is that you can reduce code repetition and also separate argument validation from the work of the function with something like:
```
        if isinstance(decimals, dict):
            colnames = decimals.keys()
            decimals_iter = decimals.items()
        elif isinstance(decimals, int):
            colnames = self.colnames
            decimals_iter = itertools.repeat(decimals)
        else:
            raise TypeError("'decimals' argument must be an int or a dict")

        for colname, decimals in zip(colnames, decimals_iter):
            col = self.columns[colname]
            ...
```